x-pack/solutions/observability/plugins/slo/server/services/find_slo_definitions.ts

```diff
@@ -10,25 +10,55 @@ import type {
   FindSLODefinitionsResponse,
   Pagination,
 } from '@kbn/slo-schema';
-import { findSloDefinitionsResponseSchema } from '@kbn/slo-schema';
+import { findSloDefinitionsWithHealthResponseSchema } from '@kbn/slo-schema';
+import type { IScopedClusterClient } from '@kbn/core/server';
 import { IllegalArgumentError } from '../errors';
 import type { SLORepository } from './slo_repository';
+import { GetSLOHealth } from './get_slo_health';
 
 const MAX_PER_PAGE = 1000;
 const DEFAULT_PER_PAGE = 100;
 const DEFAULT_PAGE = 1;
 
 export class FindSLODefinitions {
-  constructor(private repository: SLORepository) {}
+  constructor(
+    private repository: SLORepository,
+    private scopedClusterClient: IScopedClusterClient
+  ) {}
 
   public async execute(params: FindSLODefinitionsParams): Promise<FindSLODefinitionsResponse> {
     const requestTags: string[] = params.tags?.split(',') ?? [];
 
     const result = await this.repository.search(params.search ?? '', toPagination(params), {
-      includeOutdatedOnly: params.includeOutdatedOnly === true,
+      includeOutdatedOnly: !!params.includeOutdatedOnly,
       tags: requestTags,
     });
-    return findSloDefinitionsResponseSchema.encode(result);
+
+    const getSLOHealth = new GetSLOHealth(this.scopedClusterClient);
+
+    const healthResponses = await getSLOHealth.execute({
+      list: result.results.map((item) => ({
+        sloId: item.id,
+        sloInstanceId: '*',
+        sloRevision: item.revision,
+        sloName: item.name,
+      })),
+    });
+
+    const resultsWithHealth = result.results.map((slo) => {
+      const healthInfo = healthResponses.data.find((health) => health.sloId === slo.id);
+      return {
+        ...slo,
+        health: healthInfo ? (healthInfo.health as unknown) : undefined,
+      };
+    });
+
+    return findSloDefinitionsWithHealthResponseSchema.encode({
+      page: result.page,
+      perPage: result.perPage,
+      total: result.total,
+      results: params.getHealth ? resultsWithHealth : result.results,
```

Comment:
Can you add a unit test ensuring when getHealth=true, the endpoint returns the WithHealth response shape and contains the health property?